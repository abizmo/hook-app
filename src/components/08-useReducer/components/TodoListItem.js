import React from 'react'
import PropTypes from 'prop-types'

const TodoListItem = ({done, id, onDeleteTodo, onToggleTodo, todo}) => (
  <div key={id} className="input-group mb-3">
    <div className="input-group-text">
      <input
        aria-label="Checkbox to mark following todo"
        checked={done}
        className="form-check-input mt-0"
        onChange={() => onToggleTodo(id)}
        type="checkbox"
      />
    </div>
    <span
      className={`input-group-text bg-white flex-grow-1 ${done && 'text-decoration-line-through'}`}
    >{todo}</span>
    <button
      type="button"
      className="btn btn-outline-danger"
      onClick={() => onDeleteTodo(id)}
    >Borrar</button>
  </div>
)

TodoListItem.propTypes = {
  done: PropTypes.bool.isRequired,
  id: PropTypes.number.isRequired,
  onDeleteTodo: PropTypes.func.isRequired,
  onToggleTodo: PropTypes.func.isRequired,
  todo: PropTypes.string.isRequired,
}

export default TodoListItem
