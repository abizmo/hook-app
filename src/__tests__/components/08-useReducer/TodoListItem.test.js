import { shallow } from 'enzyme'

import TodoListItem from "../../../components/08-useReducer/components/TodoListItem"
import demoTodos from '../../../fixtures/demoTodos'

const [todo] = demoTodos

describe('TodoListItem tests', () => {
  const handleDeleteTodo = jest.fn()
  const handleToggleTodo = jest.fn()

  const wrapper = shallow(
    <TodoListItem
      {...todo}
      onDeleteTodo={handleDeleteTodo}
      onToggleTodo={handleToggleTodo}
    />
  )

  test('should render and match snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })
  
  test('should render the todo info', () => {
    expect(wrapper.find('span').text().trim()).toBe(todo.todo)
    expect(wrapper.find('span').hasClass('text-decoration-line-through')).toBeTruthy()
  })

  test('should call onDeleteTodo', () => {
    wrapper.find('button').simulate('click')
    expect(handleDeleteTodo).toHaveBeenCalledWith(todo.id)
  })
  
  test('should call onToggleTodo', () => {
    wrapper.find('input').simulate('change')
    expect(handleToggleTodo).toHaveBeenCalledWith(todo.id)
  })
})
