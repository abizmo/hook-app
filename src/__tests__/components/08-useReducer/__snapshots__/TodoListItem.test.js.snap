// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoListItem tests should render and match snapshot 1`] = `
<div
  className="input-group mb-3"
  key="1"
>
  <div
    className="input-group-text"
  >
    <input
      aria-label="Checkbox to mark following todo"
      checked={true}
      className="form-check-input mt-0"
      onChange={[Function]}
      type="checkbox"
    />
  </div>
  <span
    className="input-group-text bg-white flex-grow-1 text-decoration-line-through"
  >
    Comprar pan
  </span>
  <button
    className="btn btn-outline-danger"
    onClick={[Function]}
    type="button"
  >
    Borrar
  </button>
</div>
`;
